sourcegraph/cody
# Edit: Use a tracked `originalRange` when retrying a task

## Description

https://github.com/sourcegraph/cody/assets/9516420/0f6ac95a-621c-42f2-b42c-b3dd395eaaf3

Follow up on: https://github.com/sourcegraph/cody/pull/1892#issuecomment-1829379006

We were retrying fixup tasks with the `selectionRange` - this is not suitable as this range may shrink or expand depending on the code that the LLM produces.

Instead we should use the `originalRange` - the range that is provided when the fixup task is created. We still need to track the positions of the range, as it could shift due to code being added above.

## Test plan

1. Create fixups
2. Press "retry"
3. Check retry range is correct (code is not unnecessarily deleted, unless caused by the LLM)
